internal/client/integration/integration_resource.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package integrationclient
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/transport"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/cluster/integration"
+)
+
+func New(transport *transport.Client) ClientService {
+	return &client{Client: transport}
+}
+
+type client struct {
+	*transport.Client
+}
+
+type ClientService interface {
+	ManageV1alpha1ClusterIntegrationResourceServiceCreate(
+		*integration.VmwareTanzuManageV1alpha1ClusterIntegrationCreateIntegrationRequest,
+	) (*integration.VmwareTanzuManageV1alpha1ClusterIntegrationCreateIntegrationResponse, error)
+
+	ManageV1alpha1ClusterIntegrationResourceServiceRead(
+		*integration.VmwareTanzuManageV1alpha1ClusterIntegrationFullName,
+	) (*integration.VmwareTanzuManageV1alpha1ClusterIntegrationGetIntegrationResponse, error)
+
+	// ManageV1alpha1ClusterIntegrationResourceServiceUpdate has not been implemented at the API level yet.
+	//
+	// for example:
+	//   [ERROR] vertex "tanzu-mission-control_integration.default[\"tanzu-service-mesh\"]"
+	//     error: Unable to update Tanzu Mission Control integration entry,
+	//     name: tanzu-service-mesh: PUT request failed with status : 501 Not Implemented,
+	//     response: {"error":"Not Implemented: please try again later","code":12,"message":"Not Implemented: please try again later"}
+	ManageV1alpha1ClusterIntegrationResourceServiceUpdate(
```

Comment:
if it is not implemented, then can you please remove this method and also in the resource, for in-place update you can return error as not supported